captum/attr/_utils/batching.py

```diff
@@ -125,9 +125,11 @@ def _batched_generator(
         yield inputs, additional_forward_args, target_ind
     else:
         for current_total in range(0, num_examples, internal_batch_size):
-            yield _tuple_splice_range(
-                inputs, current_total, current_total + internal_batch_size
-            ), _tuple_splice_range(
+            with torch.autograd.set_grad_enabled(True):
+                inputs_splice = _tuple_splice_range(
+                    inputs, current_total, current_total + internal_batch_size
+                )
+            yield inputs_splice, _tuple_splice_range(
                 additional_forward_args,
```

Comment:
why is this not included in the context?

Reply:
This will happen in a general case when people call attribute in a `with no_grad()` block. IG works normally since we activate grads internally but slicing fails.